core/base-service/base-non-memory-caching.js

```diff
@@ -1,74 +0,0 @@
-'use strict'
-
-const makeBadge = require('../../badge-maker/lib/make-badge')
-const BaseService = require('./base')
-const { MetricHelper } = require('./metric-helper')
-const { setCacheHeaders } = require('./cache-headers')
-const { makeSend } = require('./legacy-result-sender')
-const coalesceBadge = require('./coalesce-badge')
-const { prepareRoute, namedParamsForMatch } = require('./route')
-
-// Badges are subject to two independent types of caching: in-memory and
-// downstream.
-//
-// Services deriving from `NonMemoryCachingBaseService` are not cached in
-// memory on the server. This means that each request that hits the server
-// triggers another call to the handler. When using badges for server
-// diagnostics, that's useful!
-//
-// In contrast, The `handle()` function of most other `BaseService`
-// subclasses is wrapped in onboard, in-memory caching. See `lib /request-
-// handler.js` and `BaseService.prototype.register()`.
-//
-// All services, including those extending NonMemoryCachingBaseServices, may
-// be cached _downstream_. This is governed by cache headers, which are
-// configured by the service, the user's request, and the server's default
-// cache length.
-module.exports = class NonMemoryCachingBaseService extends BaseService {
```

Comment:
This class only existed to dodge the cache behavior which is being removed herewith. The two services using this can now use BaseService directly.